src/format/packed/mod.rs

```diff
@@ -51,18 +42,72 @@ pub enum Variant {
 #[cfg(debug_assertions)]
 impl<T: Element> ::format::Validate for Packed<T> {
     fn validate(&self) {
-        assert_eq!(self.values.len(), storage!(self.size));
+        assert_eq!(self.values.len(), triangular_elems(self.size));
     }
 }
 
 size!(Packed, size, size);
 
+/// Computes the offset into the array for the upper triangular representation
+/// Copied from http://www.netlib.org/lapack/lug/node123.html
+#[inline]
+pub fn upper_triangular_offset(n: usize, i: usize, j: usize) -> usize {
+    debug_assert!(i <= j);
+    debug_assert!(i <= n);
+    debug_assert!(j <= n);
+    // Source material uses 1-based indexing
+    let i = i + 1;
+    let j = j + 1;
+    i + j * (j - 1) / 2 - 1
+}
+
+/// Computes the offset into the array for the lower triangular representation
+/// Copied from http://www.netlib.org/lapack/lug/node123.html
+#[inline]
+pub fn lower_triangular_offset(n: usize, i: usize, j: usize) -> usize {
+    debug_assert!(j <= i);
+    debug_assert!(i <= n);
+    debug_assert!(j <= n);
+    // Source material uses 1-based indexing
+    let i = i + 1;
+    let j = j + 1;
+    i + (2 * n - j) * (j - 1) / 2 - 1
+}
+
+/// Computes how many unique elements are in a symmetric square matrix of size nxn
```

Comment:
Change to “Compute the number of elements on and above or below the diagonal of a square matrix.”